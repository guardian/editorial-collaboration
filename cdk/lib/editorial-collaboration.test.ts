import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { EditorialCollaboration } from './editorial-collaboration';

describe('The EditorialCollaboration stack', () => {
	it('matches the snapshot', () => {
		const app = new App();
		const stack = new EditorialCollaboration(app, 'EditorialCollaboration', {
			stack: 'composer',
			stage: 'TEST',
			domainName: 'test.dev-gutools.co.uk',
		});
		const template = Template.fromStack(stack);
		expect(template.toJSON()).toMatchSnapshot();
	});
});
