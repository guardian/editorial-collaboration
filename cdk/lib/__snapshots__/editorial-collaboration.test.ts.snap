// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The EditorialCollaboration stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuVpcParameter",
      "GuSubnetListParameter",
      "GuSubnetListParameter",
      "GuEc2App",
      "GuCertificate",
      "GuInstanceRole",
      "GuSsmSshPolicy",
      "GuDescribeEC2Policy",
      "GuLoggingStreamNameParameter",
      "GuLogShippingPolicy",
      "GuDistributionBucketParameter",
      "GuGetDistributablePolicy",
      "GuParameterStoreReadPolicy",
      "GuAmiParameter",
      "GuHttpsEgressSecurityGroup",
      "GuWazuhAccess",
      "GuAutoScalingGroup",
      "GuApplicationLoadBalancer",
      "GuApplicationTargetGroup",
      "GuHttpsApplicationListener",
      "GuCname",
      "GuSecurityGroup",
      "GuSubnetListParameter",
      "GuDatabaseInstance",
    ],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "LoadBalancerEditorialcollaborationDnsName": {
      "Description": "DNS entry for LoadBalancerEditorialcollaboration",
      "Value": {
        "Fn::GetAtt": [
          "LoadBalancerEditorialcollaboration1076AB1D",
          "DNSName",
        ],
      },
    },
  },
  "Parameters": {
    "AMIEditorialcollaboration": {
      "Description": "Amazon Machine Image ID for the app editorial-collaboration. Use this in conjunction with AMIgo to keep AMIs up to date.",
      "Type": "AWS::EC2::Image::Id",
    },
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "LoggingStreamName": {
      "Default": "/account/services/logging.stream.name",
      "Description": "SSM parameter containing the Name (not ARN) on the kinesis stream",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "PrivateSubnets": {
      "Default": "/account/vpc/primary/subnets/private",
      "Description": "A list of private subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
    "VpcId": {
      "Default": "/account/vpc/primary/id",
      "Description": "Virtual Private Cloud to run EC2 instances within. Should NOT be the account default VPC.",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::VPC::Id>",
    },
    "editorialcollaborationPrivateSubnets": {
      "Default": "/account/vpc/primary/subnets/private",
      "Description": "A list of private subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
    "editorialcollaborationPublicSubnets": {
      "Default": "/account/vpc/primary/subnets/public",
      "Description": "A list of public subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
  },
  "Resources": {
    "AutoScalingGroupEditorialcollaborationASGB5FEA851": {
      "Properties": {
        "HealthCheckGracePeriod": 120,
        "HealthCheckType": "ELB",
        "LaunchTemplate": {
          "LaunchTemplateId": {
            "Ref": "composerTESTeditorialcollaborationF4717F5D",
          },
          "Version": {
            "Fn::GetAtt": [
              "composerTESTeditorialcollaborationF4717F5D",
              "LatestVersionNumber",
            ],
          },
        },
        "MaxSize": "2",
        "MinSize": "1",
        "Tags": [
          {
            "Key": "App",
            "PropagateAtLaunch": true,
            "Value": "editorial-collaboration",
          },
          {
            "Key": "gu:cdk:version",
            "PropagateAtLaunch": true,
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "PropagateAtLaunch": true,
            "Value": "guardian/editorial-collaboration",
          },
          {
            "Key": "LogKinesisStreamName",
            "PropagateAtLaunch": true,
            "Value": {
              "Ref": "LoggingStreamName",
            },
          },
          {
            "Key": "Stack",
            "PropagateAtLaunch": true,
            "Value": "composer",
          },
          {
            "Key": "Stage",
            "PropagateAtLaunch": true,
            "Value": "TEST",
          },
          {
            "Key": "SystemdUnit",
            "PropagateAtLaunch": true,
            "Value": "editorial-collaboration.service",
          },
        ],
        "TargetGroupARNs": [
          {
            "Ref": "TargetGroupEditorialcollaboration3D1849DE",
          },
        ],
        "VPCZoneIdentifier": {
          "Ref": "editorialcollaborationPrivateSubnets",
        },
      },
      "Type": "AWS::AutoScaling::AutoScalingGroup",
    },
    "CertificateEditorialcollaborationA30F5B14": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "DomainName": "test.dev-gutools.co.uk",
        "Tags": [
          {
            "Key": "App",
            "Value": "editorial-collaboration",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/editorial-collaboration",
          },
          {
            "Key": "Name",
            "Value": "EditorialCollaboration/CertificateEditorialcollaboration",
          },
          {
            "Key": "Stack",
            "Value": "composer",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "ValidationMethod": "DNS",
      },
      "Type": "AWS::CertificateManager::Certificate",
      "UpdateReplacePolicy": "Retain",
    },
    "DBSecurityGroupEditorialcollaborationDF4493EF": {
      "Properties": {
        "GroupDescription": "Allow traffic from EC2 instance to DB",
        "SecurityGroupEgress": [
          {
            "CidrIp": "255.255.255.255/32",
            "Description": "Disallow all traffic",
            "FromPort": 252,
            "IpProtocol": "icmp",
            "ToPort": 86,
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "editorial-collaboration",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/editorial-collaboration",
          },
          {
            "Key": "Stack",
            "Value": "composer",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "DBSecurityGroupEditorialcollaborationfromEditorialCollaborationGuHttpsEgressSecurityGroupEditorialcollaborationB5107FA954322A46C402": {
      "Properties": {
        "Description": "Allow connections from EC2 instance to DB",
        "FromPort": 5432,
        "GroupId": {
          "Fn::GetAtt": [
            "DBSecurityGroupEditorialcollaborationDF4493EF",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "GuHttpsEgressSecurityGroupEditorialcollaborationB5EB9593",
            "GroupId",
          ],
        },
        "ToPort": 5432,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "DBSecurityGroupEditorialcollaborationfromEditorialCollaborationWazuhSecurityGroup4E797D24543266AFA3F2": {
      "Properties": {
        "Description": "Allow connections from EC2 instance to DB",
        "FromPort": 5432,
        "GroupId": {
          "Fn::GetAtt": [
            "DBSecurityGroupEditorialcollaborationDF4493EF",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "WazuhSecurityGroup",
            "GroupId",
          ],
        },
        "ToPort": 5432,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "DBSubnetGroup": {
      "Properties": {
        "DBSubnetGroupDescription": "Subnet for editorial collaboration database",
        "SubnetIds": {
          "Ref": "PrivateSubnets",
        },
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/editorial-collaboration",
          },
          {
            "Key": "Stack",
            "Value": "composer",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::RDS::DBSubnetGroup",
    },
    "DatabaseEditorialcollaborationdbCCA6A1A0": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AllocatedStorage": "150",
        "AllowMajorVersionUpgrade": false,
        "AutoMinorVersionUpgrade": true,
        "CopyTagsToSnapshot": true,
        "DBInstanceClass": "db.t4g.micro",
        "DBInstanceIdentifier": "editorial-collaboration-db-test",
        "DBSubnetGroupName": {
          "Ref": "DBSubnetGroup",
        },
        "DeleteAutomatedBackups": false,
        "DeletionProtection": true,
        "EnableIAMDatabaseAuthentication": true,
        "Engine": "postgres",
        "EngineVersion": "16.2",
        "MasterUserPassword": {
          "Fn::Join": [
            "",
            [
              "{{resolve:secretsmanager:",
              {
                "Ref": "EditorialCollaborationDatabaseEditorialcollaborationdbSecretD7DDFB6E3fdaad7efa858a3daf9490cf0a702aeb",
              },
              ":SecretString:password::}}",
            ],
          ],
        },
        "MasterUsername": "editorial_collaboration",
        "MultiAZ": false,
        "Port": "5432",
        "PreferredMaintenanceWindow": "Mon:06:30-Mon:07:00",
        "PubliclyAccessible": false,
        "StorageEncrypted": true,
        "StorageType": "gp2",
        "Tags": [
          {
            "Key": "App",
            "Value": "editorial-collaboration-db",
          },
          {
            "Key": "devx-backup-enabled",
            "Value": "true",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/editorial-collaboration",
          },
          {
            "Key": "Stack",
            "Value": "composer",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VPCSecurityGroups": [
          {
            "Fn::GetAtt": [
              "DBSecurityGroupEditorialcollaborationDF4493EF",
              "GroupId",
            ],
          },
        ],
      },
      "Type": "AWS::RDS::DBInstance",
      "UpdateReplacePolicy": "Retain",
    },
    "DatabaseEditorialcollaborationdbSecretAttachment6AF09E4C": {
      "Properties": {
        "SecretId": {
          "Ref": "EditorialCollaborationDatabaseEditorialcollaborationdbSecretD7DDFB6E3fdaad7efa858a3daf9490cf0a702aeb",
        },
        "TargetId": {
          "Ref": "DatabaseEditorialcollaborationdbCCA6A1A0",
        },
        "TargetType": "AWS::RDS::DBInstance",
      },
      "Type": "AWS::SecretsManager::SecretTargetAttachment",
    },
    "DescribeEC2PolicyFF5F9295": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "autoscaling:DescribeAutoScalingInstances",
                "autoscaling:DescribeAutoScalingGroups",
                "ec2:DescribeTags",
                "ec2:DescribeInstances",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "describe-ec2-policy",
        "Roles": [
          {
            "Ref": "InstanceRoleEditorialcollaborationAEA7F45C",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "DnsRecord": {
      "Properties": {
        "Name": "test.dev-gutools.co.uk",
        "RecordType": "CNAME",
        "ResourceRecords": [
          {
            "Fn::GetAtt": [
              "LoadBalancerEditorialcollaboration1076AB1D",
              "DNSName",
            ],
          },
        ],
        "Stage": "TEST",
        "TTL": 3600,
      },
      "Type": "Guardian::DNS::RecordSet",
    },
    "EditorialCollaborationDatabaseEditorialcollaborationdbSecretD7DDFB6E3fdaad7efa858a3daf9490cf0a702aeb": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Description": {
          "Fn::Join": [
            "",
            [
              "Generated by the CDK for stack: ",
              {
                "Ref": "AWS::StackName",
              },
            ],
          ],
        },
        "GenerateSecretString": {
          "ExcludeCharacters": " %+~\`#$&*()|[]{}:;<>?!'/@"\\",
          "GenerateStringKey": "password",
          "PasswordLength": 30,
          "SecretStringTemplate": "{"username":"editorial_collaboration"}",
        },
        "Tags": [
          {
            "Key": "App",
            "Value": "editorial-collaboration-db",
          },
          {
            "Key": "devx-backup-enabled",
            "Value": "true",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/editorial-collaboration",
          },
          {
            "Key": "Stack",
            "Value": "composer",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::SecretsManager::Secret",
      "UpdateReplacePolicy": "Delete",
    },
    "GetDistributablePolicyEditorialcollaboration7454A93E": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:s3:::",
                    {
                      "Ref": "DistributionBucketName",
                    },
                    "/composer/TEST/editorial-collaboration/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GetDistributablePolicyEditorialcollaboration7454A93E",
        "Roles": [
          {
            "Ref": "InstanceRoleEditorialcollaborationAEA7F45C",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GuHttpsEgressSecurityGroupEditorialcollaborationB5EB9593": {
      "Properties": {
        "GroupDescription": "Allow all outbound HTTPS traffic",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound HTTPS traffic",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "editorial-collaboration",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/editorial-collaboration",
          },
          {
            "Key": "Stack",
            "Value": "composer",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "GuHttpsEgressSecurityGroupEditorialcollaborationfromEditorialCollaborationLoadBalancerEditorialcollaborationSecurityGroupDDE175C130004452BE66": {
      "Properties": {
        "Description": "Load balancer to target",
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "GuHttpsEgressSecurityGroupEditorialcollaborationB5EB9593",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "LoadBalancerEditorialcollaborationSecurityGroup43347B7A",
            "GroupId",
          ],
        },
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "GuHttpsEgressSecurityGroupEditorialcollaborationtoEditorialCollaborationDBSecurityGroupEditorialcollaboration37E069605432BBE81FC9": {
      "Properties": {
        "Description": "Allow connections from EC2 instance to DB",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "DBSecurityGroupEditorialcollaborationDF4493EF",
            "GroupId",
          ],
        },
        "FromPort": 5432,
        "GroupId": {
          "Fn::GetAtt": [
            "GuHttpsEgressSecurityGroupEditorialcollaborationB5EB9593",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 5432,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "GuLogShippingPolicy981BFE5A": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "kinesis:Describe*",
                "kinesis:Put*",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:kinesis:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":stream/",
                    {
                      "Ref": "LoggingStreamName",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GuLogShippingPolicy981BFE5A",
        "Roles": [
          {
            "Ref": "InstanceRoleEditorialcollaborationAEA7F45C",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "InstanceRoleEditorialcollaborationAEA7F45C": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "ec2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Path": "/",
        "Tags": [
          {
            "Key": "App",
            "Value": "editorial-collaboration",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/editorial-collaboration",
          },
          {
            "Key": "Stack",
            "Value": "composer",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ListenerEditorialcollaboration57CA1A1B": {
      "Properties": {
        "Certificates": [
          {
            "CertificateArn": {
              "Ref": "CertificateEditorialcollaborationA30F5B14",
            },
          },
        ],
        "DefaultActions": [
          {
            "TargetGroupArn": {
              "Ref": "TargetGroupEditorialcollaboration3D1849DE",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": {
          "Ref": "LoadBalancerEditorialcollaboration1076AB1D",
        },
        "Port": 443,
        "Protocol": "HTTPS",
        "SslPolicy": "ELBSecurityPolicy-TLS13-1-2-2021-06",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "LoadBalancerEditorialcollaboration1076AB1D": {
      "Properties": {
        "LoadBalancerAttributes": [
          {
            "Key": "deletion_protection.enabled",
            "Value": "true",
          },
          {
            "Key": "routing.http.x_amzn_tls_version_and_cipher_suite.enabled",
            "Value": "true",
          },
          {
            "Key": "routing.http.drop_invalid_header_fields.enabled",
            "Value": "true",
          },
        ],
        "Scheme": "internet-facing",
        "SecurityGroups": [
          {
            "Fn::GetAtt": [
              "LoadBalancerEditorialcollaborationSecurityGroup43347B7A",
              "GroupId",
            ],
          },
        ],
        "Subnets": {
          "Ref": "editorialcollaborationPublicSubnets",
        },
        "Tags": [
          {
            "Key": "App",
            "Value": "editorial-collaboration",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/editorial-collaboration",
          },
          {
            "Key": "Stack",
            "Value": "composer",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "Type": "application",
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "LoadBalancerEditorialcollaborationSecurityGroup43347B7A": {
      "Properties": {
        "GroupDescription": "Automatically created Security Group for ELB EditorialCollaborationLoadBalancerEditorialcollaboration3FFECFB4",
        "SecurityGroupIngress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow from anyone on port 443",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "editorial-collaboration",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/editorial-collaboration",
          },
          {
            "Key": "Stack",
            "Value": "composer",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "LoadBalancerEditorialcollaborationSecurityGrouptoEditorialCollaborationGuHttpsEgressSecurityGroupEditorialcollaborationB5107FA930004176ADBB": {
      "Properties": {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "GuHttpsEgressSecurityGroupEditorialcollaborationB5EB9593",
            "GroupId",
          ],
        },
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "LoadBalancerEditorialcollaborationSecurityGroup43347B7A",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "LoadBalancerEditorialcollaborationSecurityGrouptoEditorialCollaborationWazuhSecurityGroup4E797D2430004CD7EF50": {
      "Properties": {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "WazuhSecurityGroup",
            "GroupId",
          ],
        },
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "LoadBalancerEditorialcollaborationSecurityGroup43347B7A",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "ParameterStoreReadEditorialcollaborationF9082F05": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/composer/editorial-collaboration",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/composer/editorial-collaboration/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "parameter-store-read-policy",
        "Roles": [
          {
            "Ref": "InstanceRoleEditorialcollaborationAEA7F45C",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SsmSshPolicy4CFC977E": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "ec2messages:AcknowledgeMessage",
                "ec2messages:DeleteMessage",
                "ec2messages:FailMessage",
                "ec2messages:GetEndpoint",
                "ec2messages:GetMessages",
                "ec2messages:SendReply",
                "ssm:UpdateInstanceInformation",
                "ssm:ListInstanceAssociations",
                "ssm:DescribeInstanceProperties",
                "ssm:DescribeDocumentParameters",
                "ssmmessages:CreateControlChannel",
                "ssmmessages:CreateDataChannel",
                "ssmmessages:OpenControlChannel",
                "ssmmessages:OpenDataChannel",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ssm-ssh-policy",
        "Roles": [
          {
            "Ref": "InstanceRoleEditorialcollaborationAEA7F45C",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TargetGroupEditorialcollaboration3D1849DE": {
      "Properties": {
        "HealthCheckIntervalSeconds": 10,
        "HealthCheckPath": "/healthcheck",
        "HealthCheckProtocol": "HTTP",
        "HealthCheckTimeoutSeconds": 5,
        "HealthyThresholdCount": 5,
        "Port": 3000,
        "Protocol": "HTTP",
        "Tags": [
          {
            "Key": "App",
            "Value": "editorial-collaboration",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/editorial-collaboration",
          },
          {
            "Key": "Stack",
            "Value": "composer",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "TargetGroupAttributes": [
          {
            "Key": "deregistration_delay.timeout_seconds",
            "Value": "30",
          },
          {
            "Key": "stickiness.enabled",
            "Value": "false",
          },
        ],
        "TargetType": "instance",
        "UnhealthyThresholdCount": 2,
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
    "WazuhSecurityGroup": {
      "Properties": {
        "GroupDescription": "Allow outbound traffic from wazuh agent to manager",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh event logging",
            "FromPort": 1514,
            "IpProtocol": "tcp",
            "ToPort": 1514,
          },
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh agent registration",
            "FromPort": 1515,
            "IpProtocol": "tcp",
            "ToPort": 1515,
          },
        ],
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/editorial-collaboration",
          },
          {
            "Key": "Stack",
            "Value": "composer",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "WazuhSecurityGroupfromEditorialCollaborationLoadBalancerEditorialcollaborationSecurityGroupDDE175C1300066A87BFB": {
      "Properties": {
        "Description": "Load balancer to target",
        "FromPort": 3000,
        "GroupId": {
          "Fn::GetAtt": [
            "WazuhSecurityGroup",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "LoadBalancerEditorialcollaborationSecurityGroup43347B7A",
            "GroupId",
          ],
        },
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "WazuhSecurityGrouptoEditorialCollaborationDBSecurityGroupEditorialcollaboration37E069605432CE28DD4A": {
      "Properties": {
        "Description": "Allow connections from EC2 instance to DB",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "DBSecurityGroupEditorialcollaborationDF4493EF",
            "GroupId",
          ],
        },
        "FromPort": 5432,
        "GroupId": {
          "Fn::GetAtt": [
            "WazuhSecurityGroup",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 5432,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "composerTESTeditorialcollaborationF4717F5D": {
      "DependsOn": [
        "InstanceRoleEditorialcollaborationAEA7F45C",
      ],
      "Properties": {
        "LaunchTemplateData": {
          "IamInstanceProfile": {
            "Arn": {
              "Fn::GetAtt": [
                "composerTESTeditorialcollaborationProfile9C8B9B4F",
                "Arn",
              ],
            },
          },
          "ImageId": {
            "Ref": "AMIEditorialcollaboration",
          },
          "InstanceType": "t4g.micro",
          "MetadataOptions": {
            "HttpTokens": "required",
          },
          "SecurityGroupIds": [
            {
              "Fn::GetAtt": [
                "GuHttpsEgressSecurityGroupEditorialcollaborationB5EB9593",
                "GroupId",
              ],
            },
            {
              "Fn::GetAtt": [
                "WazuhSecurityGroup",
                "GroupId",
              ],
            },
          ],
          "TagSpecifications": [
            {
              "ResourceType": "instance",
              "Tags": [
                {
                  "Key": "App",
                  "Value": "editorial-collaboration",
                },
                {
                  "Key": "gu:cdk:version",
                  "Value": "TEST",
                },
                {
                  "Key": "gu:repo",
                  "Value": "guardian/editorial-collaboration",
                },
                {
                  "Key": "Name",
                  "Value": "EditorialCollaboration/composer-TEST-editorial-collaboration",
                },
                {
                  "Key": "Stack",
                  "Value": "composer",
                },
                {
                  "Key": "Stage",
                  "Value": "TEST",
                },
              ],
            },
            {
              "ResourceType": "volume",
              "Tags": [
                {
                  "Key": "App",
                  "Value": "editorial-collaboration",
                },
                {
                  "Key": "gu:cdk:version",
                  "Value": "TEST",
                },
                {
                  "Key": "gu:repo",
                  "Value": "guardian/editorial-collaboration",
                },
                {
                  "Key": "Name",
                  "Value": "EditorialCollaboration/composer-TEST-editorial-collaboration",
                },
                {
                  "Key": "Stack",
                  "Value": "composer",
                },
                {
                  "Key": "Stage",
                  "Value": "TEST",
                },
              ],
            },
          ],
          "UserData": {
            "Fn::Base64": {
              "Fn::Join": [
                "",
                [
                  "#!/bin/bash
mkdir -p $(dirname '/editorial-collaboration/main.js')
aws s3 cp 's3://",
                  {
                    "Ref": "DistributionBucketName",
                  },
                  "/composer/TEST/editorial-collaboration/main.js' '/editorial-collaboration/main.js'

# add user
groupadd editorial-collaboration
useradd -r -s /usr/bin/nologin -g editorial-collaboration editorial-collaboration
# write out systemd file
cat >/etc/systemd/system/editorial-collaboration.service <<EOL
[Service]
ExecStart=/usr/bin/node /editorial-collaboration/main.js
Restart=always
StandardOutput=journal
StandardError=journal
Environment=STAGE=TEST
User=editorial-collaboration
Group=editorial-collaboration
[Install]
WantedBy=multi-user.target
EOL
# RUN
systemctl enable editorial-collaboration
systemctl start editorial-collaboration
					",
                ],
              ],
            },
          },
        },
        "TagSpecifications": [
          {
            "ResourceType": "launch-template",
            "Tags": [
              {
                "Key": "App",
                "Value": "editorial-collaboration",
              },
              {
                "Key": "gu:cdk:version",
                "Value": "TEST",
              },
              {
                "Key": "gu:repo",
                "Value": "guardian/editorial-collaboration",
              },
              {
                "Key": "Name",
                "Value": "EditorialCollaboration/composer-TEST-editorial-collaboration",
              },
              {
                "Key": "Stack",
                "Value": "composer",
              },
              {
                "Key": "Stage",
                "Value": "TEST",
              },
            ],
          },
        ],
      },
      "Type": "AWS::EC2::LaunchTemplate",
    },
    "composerTESTeditorialcollaborationProfile9C8B9B4F": {
      "Properties": {
        "Roles": [
          {
            "Ref": "InstanceRoleEditorialcollaborationAEA7F45C",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
  },
}
`;
